scripts-components/in-view/src/index.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, ComponentInitializationError } from '@mangoweb/scripts-base'
+import { IsInView, isInView, isOutOfView, isSupported } from 'isinview'
+
+interface InViewData {
+	targets?: string
+	threshold?: number
+
+	isAboveViewClass?: string
+	isInViewClass?: string
+	isBelowViewClass?: string
+	isSeenClass?: string
+}
+
+export class InView extends Component<InViewData> {
+	static componentName = 'InView'
+
+	protected readonly targets: IsInView.Target
+	protected readonly options: Partial<IsInView.Options> = {}
+
+	protected readonly isAboveViewClass: string
+	protected readonly isInViewClass: string
+	protected readonly isBelowViewClass: string
+	protected readonly isSeenClass: string
+
+	public constructor(el: HTMLElement, data: InViewData) {
+		super(el, data)
+
+		if (!isSupported()) {
+			throw new ComponentInitializationError('InView component not supported. Add IntersectionObserver polyfill.')
+		}
+
+		this.targets = this.data.targets ? this.getChildren(this.data.targets) : this.el
+
+		if (this.data.threshold) {
+			this.options.threshold = this.data.threshold
+		}
+
+		this.isAboveViewClass = this.getProp('isAboveViewClass', 'is-aboveView')
+		this.isInViewClass = this.getProp('isInViewClass', 'is-inView')
+		this.isBelowViewClass = this.getProp('isBelowViewClass', 'is-belowView')
+		this.isSeenClass = this.getProp('isSeenClass', 'is-seen')
+	}
+
+	public init() {
+		isInView(
+			this.targets,
+			(target, details) => {
+				target.classList.add(this.isSeenClass)
```

Comment:
Tohle je možná trochu pedantické, ale já bych si držel nějakou `isSeen` `boolean`, ať se tu úplně pořád nesahá na DOM.

Reply:
Není tu DOM potřeba právě kvůli přidání CSS třídy a následnému nastylování daného stavu?

Reply:
Já si to extra sahání na DOM ospravedlnil tím, že žádné reflow a podobné náročné operace to stejně netriggruje a aspoň kód zůstává jednoduchý, přehledný.

Jak by sis ukládal `isSeen`? Přes nějakou mapu `<Element, boolean>`?

Reply:
Ping @JirkaVebr

@enzy, jak to myslíš? Bez DOMu by neměla InView komponenta význam.

Reply:
Hmm, je pravda, že mapa by přidala zbytečnou komplexitu. Pardon, nechme to takhle.